migrations/scripts/generate_schema

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Generate the converter's base intermediate database migration file from
+# the core database state and YAML configuration in schema.yml
+# Invoke from core root directory as `./migrations/scripts/generate_schema`
+# It accepts an optional command line argument for the output file path which
+# overrides the path configured in schema.yml
+
+puts "Loading application..."
+require_relative "../../config/environment"
+
+module Migrations
+  class GenerateSchema
+    def initialize(opts = {})
+      config = YAML.load_file(File.join(__dir__, "schema.yml"), symbolize_names: true)
+
+      @core_db_connection = ActiveRecord::Base.connection
+      @output_stream = StringIO.new
```

Comment:
1. It's not a big file, so I wouldn't worry about disk. Also the operating system manages the flushes to disk anyway.
2. If you wanted the flexibility, you could also initialize `@output_stream` with either `File.open` or `STDOUT` based on some parameter. They are both `IO` and should work the same.

Reply:
> 1. It's not a big file, so I wouldn't worry about disk. Also the operating system manages the flushes to disk anyway.

It's more of a mico-optimization thing and not about the file size. I wanted to avoid disk access on initializing the `output_stream`.  I'd assumed initializing a new `File` stream vs `StringIO` triggers disk access which is understandably slower than memory access.  It seems `File.new`/`File.open` don't and even if they do it'd be negligible.

> 2. If you wanted the flexibility, you could also initialize `@output_stream` with either `File.open` or `STDOUT` based on some parameter. They are both `IO` and should work the same.

Yes that's true!

Reply:
I tested using`File`,  it works but  requires adding support for making sure we aren't writing out a partial schema on exceptions.  I also don't like the fact for subsequent runs,  it'll truncates the existing schema very on in the process.

I think it's just going to be busywork  doing that now so, I'm leaving it.